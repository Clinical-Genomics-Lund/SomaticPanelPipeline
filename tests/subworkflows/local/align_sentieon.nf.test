nextflow_workflow {

    name "Test Workflow ALIGN_SENTIEON"
    script "subworkflows/local/align_sentieon.nf"
    workflow "ALIGN_SENTIEON"


    setup {
        
        run("CHECK_INPUT") {
            script "$projectDir/subworkflows/local/create_meta.nf"
            workflow {
                """
                input[0] = file("$projectDir/tests-data/gmslymphomav3-0/23KF00457-lymf_gmslymphomav3-0.csv")
                """
            }
        }

        run("SAMPLE") {
            script "$projectDir/subworkflows/local/sample.nf"
            workflow {
                """
                input[0] = CHECK_INPUT.out.fastq
                """
            }
        }

    }


    test("Should run without failures") {

        when {
            params {
                // define parameters here. Example:
                // outdir = "tests/results"
            }
            workflow {
                """
                // define inputs of the workflow here. Example:
                input[0] =  SAMPLE.out.fastq_trim
                input[1] =  CHECK_INPUT.out.meta
                """
            }
        }

        then {
            assert workflow.success
            assert snapshot(workflow.out).match()
        }

    }

}
