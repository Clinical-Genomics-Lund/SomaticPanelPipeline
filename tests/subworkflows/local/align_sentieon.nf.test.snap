{
    "Should run without failures": {
        "content": [
            [
                [
                    "23KF00457-lymf",
                    {
                        "id": "23HL02304-lymf",
                        "group": "23KF00457-lymf",
                        "diagnosis": "other",
                        "type": "N",
                        "clarity_sample_id": "CMD707A2651",
                        "ffpe": "",
                        "purity": false,
                        "sequencing_run": "230829_A01932_0262_AHJ75FDRX3",
                        "reads": "33966764",
                        "clarity_pool_id": "122-341042",
                        "sub": false
                    },
                    "23HL02304-lymf_N.QC:md5,a379e8772e63b1fe8ee83117ee82595d"
                ],
                [
                    "23KF00457-lymf",
                    {
                        "id": "23KF00457-lymf",
                        "group": "23KF00457-lymf",
                        "diagnosis": "other",
                        "type": "T",
                        "clarity_sample_id": "CMD707A3447",
                        "ffpe": "FFPE",
                        "purity": false,
                        "sequencing_run": "231024_A01932_0295_AHK5TYDRX3",
                        "reads": "75970669",
                        "clarity_pool_id": "122-355931",
                        "sub": false
                    },
                    "23KF00457-lymf_T.QC:md5,4631f1cc17658d0097598eff306a3441"
                ]
            ]
        ],
        "timestamp": "2023-12-14T23:08:14.047554"
    }
}